pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -88,15 +88,23 @@ func (w *watcher) Run(ctx context.Context) error {
 	defer os.RemoveAll(workingDir)
 	w.workingDir = workingDir
 
-	for _, repoCfg := range w.config.Repositories {
-		repo, err := w.cloneRepo(ctx, repoCfg)
+	repoCfgs := make(map[string]config.PipedRepository, len(w.config.Repositories))
+	for _, repo := range w.config.Repositories {
+		repoCfgs[repo.RepoID] = repo
+	}
```

Comment:
nit: We are having a helper function for this kind of job.
`w.config.GetRepositoryMap`

Reply:
well, fancy helper ;)